src/inc/fixuppointer.h

```diff
@@ -64,6 +64,20 @@ class RelativePointer
         return dac_cast<PTR_TYPE>(base + m_delta);
     }
 
+    // Returns value of the embedded encoded pointer with DPTR struct base. Assumes that the pointer is not NULL.
+    template<typename TYPE_T>
+    PTR_TYPE GetValueEmbedded(DPTR(TYPE_T) pStructBase) const
+    {
+        TYPE_T *pHostStructBase = (TYPE_T *) pStructBase;
```

Comment:
We think that this cast from DPTR(T) to T*

```cpp
TYPE_T *pHostStructBase = (TYPE_T *) pStructBase;
```

is implemented as the following

```cpp
operator type*() const
{
   return (type*)DacInstantiateTypeByAddress(this->m_addr, sizeof(type), true);
}
```

which is similar to how operator-> is implemented

```cpp
type* operator->() const
{
   return (type*)DacInstantiateTypeByAddress(this->m_addr, sizeof(type), true);
}
```

Could you, please, suggest in which cases this is incorrect and how is it better to obtain host pointer from DPTR?